weldx/geometry.py

```diff
@@ -1379,70 +1381,241 @@ def shapes(self) -> list[Shape]:
 # Trace segment classes -------------------------------------------------------
 
 
-class LinearHorizontalTraceSegment:
-    """Trace segment with a linear path and constant z-component."""
+class DynamicTraceSegment:
+    """Trace segment that can be defined by a ``SpatialSeries``."""
 
-    @UREG.wraps(None, (None, _DEFAULT_LEN_UNIT), strict=True)
-    def __init__(self, length: pint.Quantity):
-        """Construct linear horizontal trace segment.
+    def __init__(
+        self,
+        series: Union[
+            SpatialSeries, pint.Quantity, DataArray, str, MathematicalExpression
+        ],
+        max_s: float = 1,
+        limit_orientation_to_xy: bool = False,
+        **kwargs,
+    ):
+        """Initialize a `DynamicTraceSegment`.
+
+        Parameters
+        ----------
+        series:
+            A `~weldx.core.SpatialSeries` that describes the trajectory of the trace
+            segment. Alternatively, one can pass every other object that is valid as
+            first argument to of the ``__init__`` method of the
+            `~weldx.core.SpatialSeries`.
+        max_s:
+            [only expression based `~weldx.core.SpatialSeries`] The maximum value of
+            the passed series ``s`` parameter. The value defines the segments length by
+            evaluating the expression on the interval [0, ``max_s``]
+        limit_orientation_to_xy:
+            If `True`, the orientation vectors of the coordinate systems along the trace
+            are confined to the xy-plane.
+        kwargs:
+            A set of keyword arguments that will be forwarded to the ``__init__`` method
+            of the `~weldx.core.SpatialSeries` in case the ``series`` parameter isn't
+            already a `~weldx.core.SpatialSeries`.
+        """
+        if not isinstance(series, SpatialSeries):
+            series = SpatialSeries(series, **kwargs)
+
+        self._series = series
+        self._max_s = max_s
+        self._limit_orientation = limit_orientation_to_xy
+
+        if series.is_expression:
+            self._derivative = self._get_derivative_expression()
+            self._length_expr = self._get_length_expr()
+        else:
+            self._derivative = None
+            self._length_expr = None
+
+        self._length = self.get_section_length(self._max_s)
+
+    def _get_component_derivative_squared(self, i: int):
```

Comment:
can you add the return value for the type hint here @vhirtham 
Is it a sympy expression?

Reply:
I will try it. When I printed the type, I got a derived type. I have to figure out the base type